library/spdm_responder_lib/libspdm_rsp_csr.c

```diff
@@ -113,9 +113,7 @@ libspdm_return_t libspdm_get_response_csr(libspdm_context_t *spdm_context,
                                                response_size, response);
     }
 
-    opaque_data = (void*)((size_t)(spdm_request + 1));
-
-    requester_info = (void*)(opaque_data + opaque_data_length);
+    requester_info = (void*)((size_t)(spdm_request + 1));
```

Comment:
Why the cast to `size_t`?

Reply:
`const spdm_get_csr_request_t *spdm_request;`
It needs to be converted to a number first, and then to a pointer

Reply:
`(spdm_request + 1)` is already a pointer. The `(size_t)` is not needed.

Reply:
requester_info = (uint8_t *)(spdm_request + 1);
error: cast discards ‘const’ qualifier from pointer target type

Reply:
What compiler are you using?

Reply:
GCC compiler
I tried using CI to test more compilers，GCC reports error
https://github.com/Xiaohanjlll/libspdm/actions/runs/5052621802